src/cmd-buildupload

```diff
@@ -53,32 +59,37 @@ def parse_args():
 
 
 def cmd_upload_s3(args):
+    bucket, prefix = args.url.split('/', 1)
+    # New arguments to replace 'url'
+    if args.bucket and args.prefix:
```

Comment:
Sorry, I shoud've been more explicit about what I meant here. Instead of adding new CLI switches that shadow the required `url` arg, we can just do the split here, then literally add new attributes to the `args` object itself (using the `args.bucket, args.prefix = ` line in my other comment). That way, when we pass `args` to e.g. `s3_upload_build`, it already has access to them through `args.bucket` and `args.prefix`.

(Not against adding those CLI args though... but we'd need to also tweak arg parsing so that `url` is no longer required and only error out if none of `url`, `--bucket`, and `--prefix` are specified -- let's keep that separate from this PR?)

Reply:
>let's keep that separate from this PR?

Agree

Reply:
>> let's keep that separate from this PR?
>
> Agree

Hmm, still seeing the new CLI args in this patch.

> although I don't think mutating `args` is a good idea

I mostly agree, though I think it's useful for args which are wholly derived from other args (like `bucket` and `prefix` here). We shouldn't nest other random objects in there though.

Anyway, passing the bucket and keys as separate args as you had originally is completely fine too if you'd like to keep that.

Reply:
Right, removed these options. I'll add those in the follow-up PRs